src/map/utils/battleutils.cpp

```diff
@@ -2877,11 +2877,7 @@ namespace battleutils
             }
         }
 
-        // Bonus attack currently only from footwork
-        if (bonusAttPercent >= 1)
-        {
-            attack = static_cast<uint16>(attack * bonusAttPercent);
-        }
+        attack = static_cast<uint16>(attack * bonusAttPercent);
```

Comment:
May want to do this to prevent divisions by zero
```suggestion
        attack = std::max(1, static_cast<uint16>(attack * bonusAttPercent);
```

Reply:
Ill have to take a look - one of the first things i tested was footwork kick attacks with no kick attack gear -but since the attack is in the numerator couldve just caused awful dmg

Reply:
Ah ha, attack is always in the numerator so nothing will explode, just dmg drops through floor

I've updated all callers of GetDamageRatio to begin at 1.0 for flat attack, >1.0 for bonuses

Reply:
LGTM, I don't see a "resolve" button here but if you've got one then go ahead and press it.